discgen/utils.py

```diff
@@ -117,6 +124,79 @@ def create_svhn_streams(training_batch_size, monitoring_batch_size):
                           monitoring_batch_size)
 
 
+class Colorize(AgnosticSourcewiseTransformer):
+    def __init__(self, data_stream, **kwargs):
+        super(Colorize, self).__init__(
+             data_stream=data_stream,
+             produces_examples=data_stream.produces_examples,
+             **kwargs)
+
+    def transform_any_source(self, source, _):
+        iters, dim, height, width = source.shape
```

Comment:
If source.ndim == 3 for image data, would the fields then be:
```python
iters, dim, length = source.shape
```
ie: a flat packing where length = width * height? If so, agreed that would probably also be easy to support.